plotting_tools/PlotHelpers.h

```diff
@@ -0,0 +1,222 @@
+{
+  int n_bins = lxy_bins.size();
+  TH1D* hist = new TH1D((input_hist_name+"_1widthbins").c_str(), (input_hist_name+"_1widthbins").c_str(), n_bins-1, 0, n_bins-1);
+  for(int i=0; i<n_bins-1; i++)
+  {
+    hist->SetBinContent(i+1, input_hist->GetBinContent(i+1));
+  }
+  hist->GetXaxis()->SetLabelSize(0.04);
+  return hist;
+}
+
+TH1D* get_lxy_with_outer_tracker(TH1D* input_hist, string input_hist_name, vector<int> lxy_bins, float x_max, float rebin, float int_lumi, float cross_sec, float BR, float N_tot)
+{
+  float binListExtended[8] = {0, 2, 10, 24, 31, 70, 110, x_max}; // mm
+  int n_bins_max = 8;
+  int n_bins = 7;
+  int n_bins_i = 0;
+  for(int i=0; i<n_bins_max; i++)
+  {
+    float value = input_hist->GetBinContent(i+1)*rebin*int_lumi*cross_sec*BR/N_tot;
+    if( value < 0.00001) break;
+    n_bins_i++;
+  }
+  // TH1D* hist = input_hist.Clone();
+  double outer_tracker_content;
+  if(n_bins_i >= 8) {
+    outer_tracker_content = input_hist->GetBinContent(7+1);
+    n_bins = n_bins_i;
+  }
+  else {n_bins = n_bins_i+1;}
+  float binList[n_bins];
+  for(int i=0; i<n_bins_i+1; i++)
+  {
+    binList[i] = binListExtended[i];
+  }
+  if(n_bins_i < 8) {binList[n_bins_i] = binList[n_bins_i-1] + 1e-10;}
+  
+  TH1D* hist = new TH1D((input_hist_name+"_outer-tracker").c_str(), (input_hist_name+"_outer-tracker").c_str(), n_bins-1, binList);
+  for(int i=0; i<n_bins_i; i++)
+  {
+    // if(input_hist->GetBinContent(i+1) < y_max) break;
+    hist->SetBinContent(i+1, input_hist->GetBinContent(i+1));
+  }
+  if(n_bins_i >= 8) {
+    std::cout << n_bins_i << std::endl;
+    hist->SetBinContent(n_bins_i+1, outer_tracker_content);
+  }
+  if(n_bins_i < 8) {binList[n_bins_i+1] = 1e-10;}
+  hist->GetXaxis()->SetLabelSize(0.04);
+  return hist;
+}
+
+void add_overflow_bin(TH1D* hist, double xMax)
+{
+  int maxbin = hist->FindBin(xMax);
+  double content = 0;
+  for(int i=maxbin-1; i<hist->GetNbinsX()+1; i++){
+    content+=hist->GetBinContent(i);
+  }
+  // hist->GetXaxis()->SetLimits(xMin, xMax);
+  hist->SetBinContent(hist->FindBin(xMax)-1, content);
+}
+
+vector<int> rebin_histogram(TH1D* hist, string hist_name)
+{
+  vector<int> lxy_bins;
+  if (hist_name == "minlxy-muon_lxy_rebinned") {lxy_bins = {0, 2, 10, 24, 31, 70, 110};} 
+  if (hist_name == "maxlxy-muon_lxy_rebinned") {lxy_bins = {0, 2, 10, 24, 31, 70, 110};} 
+  if (hist_name == "minlxy-muon_lxy_rebinned_extended") {lxy_bins = {0, 2, 10, 24, 31, 70, 110, 1300, 3000, 7300};} 
+  if (hist_name == "minlxy-muon_lxy_rebinned_extended_gen") {lxy_bins = {0, 110, 1300, 3000, 7300};}
+  if (hist_name == "minlxy-muon_lxy_rebinned_extended_general") {lxy_bins = {0, 110, 1300, 3000, 7300};}
+  for(int i=0; i<hist->GetNbinsX(); i++){
+    hist->SetBinContent(i+1, hist->GetBinContent(i+1)/(lxy_bins[i+1]-lxy_bins[i]));
+  }
+  return lxy_bins;
+}
+
+void set_hist_layout(THStack* stack, tuple<bool, bool, bool, int, double, double, double, double, string, string> params, bool weighted)
```

Comment:
This tuple as an argument looks ridicoulous :D (not your fault of course!) A simple improvement would be to add a `typedef` and just call it something, like `HistParams` or `PlottingParams`.